src/sdk_codes.ts

```diff
@@ -158,14 +158,53 @@ export function warnAuthRetrying(
   );
 }
 
-export function warnStreamDisconnected(reason: string, logger: Logger): void {
-  logger.warn(getSdkErrMsg(5001, reason));
+// Track disconnect attempts for throttling warning logs
+let disconnectAttempts = 0;
+let lastConnectionSuccess = 0;
+
+// Restart disconnect counter when connection succeeds
+export function restartDisconnectCounter(): void {
+  disconnectAttempts = 0;
+  lastConnectionSuccess = Date.now();
 }
 
-export function warnStreamRetrying(seconds: number, logger: Logger): void {
-  logger.warn(getSdkErrMsg(5003, `${seconds}`));
+// Reset disconnect counter
+export function resetDisconnectCounter(): void {
+  disconnectAttempts = 0;
+  lastConnectionSuccess = 0;
 }
 
+
+// Combined warning function for both disconnect and retry in a single message
+export function warnStreamDisconnectedWithRetry(reason: string, ms: number, logger: Logger): void {
```

Comment:
Arguable that this logic should not be encapsulated in sdk_codes, but I think it's a small enough change as to not cause maintainability issues.